src/Sylius/Bundle/AttributeBundle/Controller/AttributeController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\AttributeBundle\Controller;
+
+use Sylius\Bundle\AttributeBundle\AttributeType\TextAttributeType;
+use Sylius\Bundle\ResourceBundle\Controller\ResourceController;
+use Sylius\Component\Attribute\Model\AttributeInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class AttributeController extends ResourceController
+{
+    /**
+     * @return Response
+     */
+    public function renderAttributeTypesAction()
+    {
+        $attributeTypes = $this->get('sylius.registry.attribute_type')->all();
+
+        return $this->render('SyliusAttributeBundle::attributeTypesModal.html.twig', array('attributeTypes' => $attributeTypes));
+    }
+
+    /**
+     * @return Response
+     */
+    public function renderAttributesAction()
+    {
+        $form = $this->get('form.factory')->create(
+            'sylius_product_attribute_choice',
+            null,
+            array(
+                'expanded' => true,
+                'multiple' => true,
+            )
+        );
+
+        return $this->render('SyliusAttributeBundle::attributeChoice.html.twig', array('form' => $form->createView()));
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function renderAttributeTypesFormsAction(Request $request)
+    {
+        $attributeRepository = $this->get('sylius.repository.product_attribute');
+        $forms = array();
+
+        $choices = ($request->query->has('sylius_product_attribute_choice')) ? $request->query->get('sylius_product_attribute_choice') : array();
+
+        foreach ($choices as $choice) {
+            $attribute = $attributeRepository->find($choice);
```

Comment:
You can find multiple attributes with single ``findBy(array('id' => $choices))`` if choices contains array of ids.